packages/plexus/src/LayoutManager/Coordinator.tsx

```diff
@@ -96,10 +102,10 @@ export default class Coordinator {
       id,
       options,
       input: { edges, vertices },
-      status: { phase: 'not-started' },
+      status: { phase: ECoordinatorPhase.NotStarted },
     };
     const isDotOnly = Boolean(options && options.useDotEdges);
-    const phase = isDotOnly ? 'dot-only' : 'positions';
+    const phase = isDotOnly ? EWorkerPhase.DotOnly : EWorkerPhase.Positions;
```

Comment:
`EWorkerPhase` was imported from a `./types` file but is being used to define values. Should `EWorkerPhase` exist as `WorkerPhases` in a constants file? I think mixing constants and types could get messy. I don't expect `EWorkerPhase` to have usable values, just an enum definition.

Reply:
The enums are types, and they define a set of values of that type. For instance, `EWorkerPhase.DotOnly` is a value of type `EWorkerPhase`.

Here are [examples of the upsides](https://www.typescriptlang.org/play/index.html#src=enum%20ThingAB%20%7B%0D%0A%20%20%20%20A%20%3D%20'A'%2C%0D%0A%20%20%20%20B%20%3D%20'B'%2C%0D%0A%7D%0D%0A%0D%0Aenum%20ThingAB2%20%7B%0D%0A%20%20%20%20A%20%3D%20'A'%2C%0D%0A%20%20%20%20B%20%3D%20'B'%2C%0D%0A%7D%0D%0A%0D%0Aconst%20one%3A%20ThingAB%20%3D%20ThingAB.A%3B%0D%0Aconst%20two%3A%20ThingAB2%20%3D%20ThingAB2.A%3B%0D%0Aconsole.log(one%20%3D%3D%3D%20two)%3B%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20always%20false%20error%20%20%0D%0Aconst%20three%3A%20ThingAB2%20%3D%20ThingAB.A%3B%20%20%20%20%20%20%2F%2F%20not%20assignable%20error%0D%0A%0D%0A):

```typescript
enum ThingAB {
    A = 'A',
    B = 'B',
}

enum ThingAB2 {
    A = 'A',
    B = 'B',
}

const one: ThingAB = ThingAB.A;
const two: ThingAB2 = ThingAB2.A;
console.log(one === two);               // always false error  
const three: ThingAB2 = ThingAB.A;      // not assignable error
```

I would say enums are pretty nice, and I've missed them in JS and flow. 

Unfortunately, unions of enums (or something akin to unions) are not possible in TypeScript (not sure about other languages), so I did some strange stuff with `EWorkerPhase` deriving values from `ECoordinatorPhase`, but otherwise I believe their use in plexus is fairly idiomatic.

The [official rationale](https://en.wikipedia.org/wiki/Enumerated_type#Rationale) for them.

Enums in [C#](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/enum#example-2), [Swift](https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html), [Java](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html), and [others](https://rosettacode.org/wiki/Enumerations).